server/src/main/java/io/crate/exceptions/ColumnUnknownException.java

```diff
@@ -35,6 +35,11 @@ public ColumnUnknownException(String columnName, RelationName relationName) {
         this.relationName = relationName;
     }
 
+    public ColumnUnknownException(String message) {
+        super(message);
+        this.relationName = null;
+    }
+
```

Comment:
I think this would cause issues if `getTableIdents` is called. Items of the iterable shouldn't be null (but instead the iterable would have to be empty)

But I'm also not sure if this is good to allow in general. The reason we return the table idents is so that we can verify whether the user has permission to know if a table/column exists or not.  So if we don't return the right relation we could leak information.

I'd suggest to change the exception hierarchy a bit and have a `UnknownObjectKey` and use that in the `SubscriptFunction`. Then `ColumnUnknownException` could extend `UnknownObjectKey`.

Reply:
Or an alternative could be a static factory method that clarifies under which circumstances it is okay to use this exception without having a relation name.